[MISC] Rewrite and update html build instructions (#1032)
bids-standard/bids-specification: CONTRIBUTING.md

We are using mkdocs to render our specification.
Please follow these instructions if you would like to build the specification locally.

#### 1. Download the BIDS specification [repository](https://github.com/bids-standard/bids-specification/tree/master) onto your computer

This can be done by clicking the green button on the right titled "Clone or
download"
or using [this link](https://github.com/bids-standard/bids-specification/archive/master.zip).

> 

Reviewer: ```suggestion

Or you can use the following git command in a terminal: `git clone https://github.com/bids-standard/bids-specification.git`.

```

#### 2. In the terminal (command line) navigate to your local version of the specification

This location will have the same files you see on our
[main specification page](https://github.com/bids-standard/bids-specification).
Note that a file browser window may not show the hidden files
(those that start with a period, like `.remarkrc`).

> 

Reviewer: ```suggestion

If you cloned the repository using the git command above, you can then just do: `cd bids-specification`.

```

To install all of this make sure you have a recent version of Python on your computer.
The [DataLad Handbook](http://handbook.datalad.org/en/latest/intro/installation.html#python-3-all-operating-systems) provides helpful instructions for setting up Python.

In general, we strongly recommend that you install all dependencies in an isolated Python environment.

> For example using `conda`, as described in the section on Python in the DataLad Handbook, linked above.

Reviewer: ```suggestion
For example using `conda`, as described in the section on Python in the DataLad Handbook, linked above.

```bash
conda create --name bids-spec
conda activate bids-spec
.```

```

> Or alternatively using `venv`, as described in this [Real Python tutorial](https://realpython.com/python-virtual-environments-a-primer/).

Reviewer: ```suggestion
Or alternatively using `venv`, as described in this [Real Python tutorial](https://realpython.com/python-virtual-environments-a-primer/).

A short version of the commands needed to create and activate your `venv` virtual environment would look like:

```
python3 -m venv env
source env/bin/activate
```

```

Reviewer: Can't seem to suggest code blocks without messing things up so I went for  ".```" for now

To install all of this make sure you have a recent version of Python on your computer.
The [DataLad Handbook](http://handbook.datalad.org/en/latest/intro/installation.html#python-3-all-operating-systems) provides helpful instructions for setting up Python.

In general, we strongly recommend that you install all dependencies in an isolated Python environment.

> For example using `conda`, as described in the section on Python in the DataLad Handbook, linked above.

Reviewer 2: ```suggestion
For example using `conda`, as described in the [DataLad Handbook](http://handbook.datalad.org/en/latest/intro/installation.html#conda).
```
However, I'm not a fan of this handbook because it is biased towards using DataLad.

Author: If you have an application agnostic introduction to Python installation that you'd rather link to here, we're open to suggestions! The DataLad Handbook just finds a nice balance between verbose (good for newcomers) and still to the point, IMHO.

Reviewer 2: I thought an explicit link would be better. But after browsing it, I didn't find it clear. Maybe [something like this one](https://geohackweek.github.io/Introductory/01-conda-tutorial/) would be better. But I am not a conda user.

Author: Thanks for the suggestion, I like the geohackweek resource for a quick `conda` intro -- it seems concise and a bit more helpful than the DataLad Handbook resource for this case. I'll add it.

```bash
conda create --name bids-spec
conda activate bids-spec
```

Or alternatively using `venv`, as described in this [Real Python tutorial](https://realpython.com/python-virtual-environments-a-primer/).

A short version of the commands needed to create and activate your `venv` virtual environment would look like:

```bash

> python3 -m venv env

Reviewer 2: ```suggestion
python -m venv env
```
I think `python3` is a Debian thing.

Author: Also on Ubuntu, but yeah - we can probably drop it: because the `python -m venv` command will only work with Python 3.4+ anyhow, so users won't accidentally create a Python 2 venv.

source env/bin/activate
```

Note that this will create a local directory called `env` within the bids-specification directory
but that its content will not be tracked by `git` because it is listed in the `.gitignore` file.

Once you have activated your isolated Python environment,
an easy way to install the correct version of mkdocs and all the other required extensions
is to use the `requirements.txt` file contained in this repository as follows:

```bash
pip install -U pip

> pip install -r requirements.txt

Reviewer 2: ```suggestion
pip install -r requirements.txt
pip install tools/schemacode/
```
Aren't we missing the `schemacode`?

Author: That's actually part of the `requirements.txt`, so I suspect your issue occurred because of a non-up-to-date pip and/or not calling `pip install -r requirements.txt` from the root of the repository (so that the relative link to schemacode works)

https://github.com/bids-standard/bids-specification/blob/ba11771e76978264445cfc35d52eda14d1ca30fc/requirements.txt#L7

Reviewer 2: I've just tested it again and this line is indeed redundant.

```

The first command ensures you are using an up to date version of `pip`,

> and the second command installs all dependencies.

Reviewer 2: Explain `pip install tools/schemacode/`, for example
> The third line installs a module with [mkdocs-macros](https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md#using-macros) used for generating some of the documentation contents.

To install all of this make sure you have a recent version of Python on your computer.
The [DataLad Handbook](http://handbook.datalad.org/en/latest/intro/installation.html#python-3-all-operating-systems) provides helpful instructions for setting up Python.

In general, we strongly recommend that you install all dependencies in an isolated Python environment.
For example using `conda`, as described in the section on Python in the DataLad Handbook, linked above.

```bash
conda create --name bids-spec
conda activate bids-spec
```

Or alternatively using `venv`, as described in this [Real Python tutorial](https://realpython.com/python-virtual-environments-a-primer/).

A short version of the commands needed to create and activate your `venv` virtual environment would look like:

```bash
python3 -m venv env
source env/bin/activate
```

Note however, that this will create a local directory called `env` within the bids-specification directory.

> Make sure not to commit that directory (your environment) when adding  and committing your changes via `git`

Reviewer: ```suggestion
Make sure not to commit that directory (your environment) when adding and committing your changes via `git`
```

Reviewer: extra space

> Make sure not to commit that directory (your environment) when adding  and committing your changes via `git`

Reviewer: ```suggestion
Note that this will create a local directory called `env` within the bids-specification directory but that its content will not be tracked by GIT because it is listed in the `.gitignore` file.
```

Author: ahh :+1: thanks!
